source/common/http/http2/codec_impl.cc

```diff
@@ -985,9 +985,15 @@ int ConnectionImpl::onInvalidFrame(int32_t stream_id, int error_code) {
     }
     break;
 
+  case NGHTTP2_ERR_FLOW_CONTROL:
+  case NGHTTP2_ERR_PROTO:
+  case NGHTTP2_ERR_STREAM_CLOSED:
+    // Known error conditions that should trigger connection close.
```

Comment:
Use of a binary fuzz input to provide test coverage is not very satisfying, but crafting two header frames for the same stream which can be sent on the same connection would require composing header frames by hand.  I don't know of tests that do that today with the possible exception of some of the http2 flood integration tests.

Reply:
This was one way I did this horrible though I agree
https://github.com/envoyproxy/envoy/blob/0fae6970ddaf93f024908ba304bbd2b34e997a51/test/integration/multiplexed_integration_test.cc#L1646

Reply:
I can probably do better here but a better version of this test would require use of Http2Frame::makeRequest and sendFrame.